daemon/containerd/service.go

```diff
@@ -0,0 +1,369 @@
+package containerd
+
+import (
+	"context"
+	"io"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/platforms"
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/backend"
+	"github.com/docker/docker/api/types/filters"
+	imagetype "github.com/docker/docker/api/types/image"
+	registrytypes "github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/builder"
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/daemon/images"
+	"github.com/docker/docker/errdefs"
+	"github.com/docker/docker/image"
+	"github.com/docker/docker/layer"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+// ImageService implements daemon.ImageService
+type ImageService struct {
```

Comment:
Linter didn't allow the type to be named `ContainerdStore` (starts with package name), and wanted to have all exported functions to have a GoDoc, so I;

- renamed it to `containerd.ImageService` (to match the `daemon.ImageService` implementation
- copied all GoDoc "as-is" from the other service (some comments may need some cleaning up)

Reply:
🚲 🏠 How about naming the package + type `github.com/docker/docker/daemon/images/c8dimage.Service`? I figure putting the package under `daemon/images` would emphasize that the scope of the package is limited to the image backend aspects of interfacing with containerd, distinct from the container lifecycle and daemon supervisor aspects in libcontainerd.